### PR TITLE
[WIP] feat: add managed model registry prometheus job, metrics, and alering rules, fixes RHOAIENG-4273

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description

Adds managed model registry prometheus config for the following:
* job
* metrics
* alering rules

Fixes RHOAIENG-4273

https://issues.redhat.com/browse/RHOAIENG-4273

## How Has This Been Tested?

WIP

## Merge criteria

- [X] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [X] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [X] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dhirajsb** [comment]
This is a draft PR with prometheus config changes to the yaml file. I'm creating another PR to enable loading this config in odh incubation. This PR should be merged after the model registry component itself is merged into rhds. 

Please review and let me know if there are any changes or anything else is needed.

File: config/monitoring/prometheus/apps/prometheus-configs.yaml [hunk 1]
```diff
@@ -1577,3 +1602,92 @@ data:
...
+          record: probe_success:burnrate6h
+
+  model-registry-operator-alerting.rules: |
+    groups:
+      - name: SLOs-probe_success_model_controller
+        rules:
+        - alert: Model Registry Operator Probe Success Burn Rate
+          annotations:
+            message: 'High error budget burn for {{ $labels.instance }} (current value: {{ $value }}).'
+            triage: "https://gitlab.cee.redhat.com/service/managed-tenants-sops/-/blob/main/RHODS/Model-Serving/rhods-model-registry-operator-probe-success-burn-rate.md"
```
Comment (zdtsw) [inline config/monitoring/prometheus/apps/prometheus-configs.yaml:1658]: one thing before merge this PR is to get https://gitlab.cee.redhat.com/service/managed-tenants-sops/-/blob/main/RHODS/Model-Serving/rhods-model-registry-operator-probe-success-burn-rate.md created.
preferably to name the file as `rhoai-model-registry-operator-probe-success-burn-rate.md`

Comment (dhirajsb) [reply config/monitoring/prometheus/apps/prometheus-configs.yaml:1658]: Related to that, I used the simple name `Model Registry Operator` in the config names and descriptions. I noticed that some components used ODH prefix. Does model registry need to use `RHOAI Model Registry Operator` instead?

Comment (zdtsw) [reply config/monitoring/prometheus/apps/prometheus-configs.yaml:1658]: think you can make a PR and ask (someone) in the project to review/approve it.
should be a owner file somewhere with name list there

Comment (zdtsw) [reply config/monitoring/prometheus/apps/prometheus-configs.yaml:1658]: tbh, i do not know the answer. I would say if ODH already exists by other component, then you can use it, or just skip ODH or any downstream name. but better check with project, if any hard/soft rules.

**Comment by dhirajsb** [comment]
@zdtsw I updated the file name etc. to match the merge request for the SOPs document. I'm waiting on the managed tenants team to review it.

**Comment by dhirajsb** [comment]
@zdtsw sops have been added.

**Comment by zdtsw** [comment]
@ykaliuta if  https://github.com/red-hat-data-services/rhods-operator/pull/329/ has  the latest changes from here, we can have https://github.com/red-hat-data-services/rhods-operator/pull/318 closed.

**Comment by ykaliuta** [comment]
> @ykaliuta if #329 has the latest changes from here, we can have #318 closed.

sure, but let's finish #329 first.

**Comment by dhirajsb** [comment]
Closing this PR since these changes were included in https://github.com/red-hat-data-services/rhods-operator/pull/329

**Pull request closed without merging**